src/services/WolfDen.API/Controllers/LeaveManagement/LeaveTypeController.cs

```diff
@@ -0,0 +1,38 @@
+﻿using MediatR;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using WolfDen.Application.Requests.Commands.LeaveManagement.LeaveTypeCommand;
+
+namespace WolfDen.API.Controllers.Leavemanagement
+{
+    [Route("api/LeaveType")]
+    [ApiController]
+    public class LeaveTypeController : ControllerBase
+    {
+        private readonly IMediator _mediator;
+
+        public LeaveTypeController(IMediator mediator)
+        {
+            _mediator = mediator;
+
+        }
+        [HttpPost("Add-New-Type")]
+        public async Task<string> AddleaveType([FromBody] LeaveTypeAddCommand command)
+        {
+            int result = await _mediator.Send(command);
+            if (result == -1)
+            {
+                return "Only Higher User can Add new leave Type";
```

Comment:
Handle this logic in the application layer

Reply:
Logic Changed to application layer